mlrun/api/utils/periodic.py

```diff
@@ -49,8 +49,9 @@ def cancel_periodic_function(name: str):
     global tasks
     logger.debug("Canceling periodic function", name=name)
     if name in tasks:
-        tasks[name].cancel()
+        task = tasks[name]
         del tasks[name]
+        task.cancel()
```

Comment:
Why ?
The idea was to remove only after it's actually canceled

Reply:
when function cancel itself it doesn't able to get to delete its record from the dictionary, because it being terminated.

Reply:
So add a comment otherwise the next person touching the code might break it